apps/client/src/views/projects/ManageEnvironments.vue

```
@@ -49,29 +49,33 @@ const cancel = () => {
 }
 
 const addEnvironment = async (environment) => {
-  cancel()
-  try {
-    await projectEnvironmentStore.addEnvironmentToProject(environment)
-  } catch (error) {
-    snackbarStore.setMessage(error?.message, 'error')
+  if (!project.value.locked) {
+    cancel()
+    try {
+      await projectEnvironmentStore.addEnvironmentToProject(environment)
+    } catch (error) {
+      snackbarStore.setMessage(error?.message, 'error')
+    }
   }
 }
 
 const deleteEnvironment = async (environment) => {
-  try {
-    await projectEnvironmentStore.deleteEnvironment(environment.id)
-  } catch (error) {
-    snackbarStore.setMessage(error?.message, 'error')
+  if (!project.value.locked) {
```

Intérêt de faire un `fail fast` et d'afficher un message dans la snackbar ?

---

le if est déjà une sécurité de plus vu que le bouton est `disabled` mais je peux rajouter le catch